### ISSUE TITLE
Binaries entering "undefined" state, not completing evolution

### ISSUE THREAD

**Issue by celiotine** [POSYDON-code/POSYDON-public#205]
There are many binaries that are not completing their evolution because they enter an "undefined" state and then immediately go to the "END" event. This only happens when S1 is a compact object, S2 is in a non-burning state ('stripped_He_non_burning'' or 'H-rich_non_burning'), and the binary is going through the CO_HMS_RLO or CO_HeMS steps:

<img width="1010" alt="Screen Shot 2023-11-29 at 1 14 24 PM" src="https://github.com/POSYDON-code/POSYDON/assets/13874913/2a62c306-3b31-4885-beff-ee58684115bf">

These binaries do not fail, but produce a warning, which is making it difficult to debug:

<img width="787" alt="Screen Shot 2023-11-29 at 1 40 51 PM" src="https://github.com/POSYDON-code/POSYDON/assets/13874913/044e8d2a-f003-44ce-a426-39132bb0c043">

I noticed that the "undefined" state is not in the flow. It is only defined in 'utils/common_functions.py' in the function `get_binary_state_and_event_and_mt_case()`. I am wondering if this is the issue. However, if you look at this function, it also seems these binaries should not reach this state in the first place, as they all are in RLO2 when this happens.

To reproduce this warning for a single binary, you can evolve the following binary in a notebook using the current development branch and get the simulation properties from the given file on Quest: 

```
sim_kwargs = simprop_kwargs_from_ini("/projects/b1119/cliotine/dev_branch_pops/params_v2.ini")
sim_prop = SimulationProperties(**sim_kwargs)
sim_prop.load_steps(verbose=True)

star_1 = SingleStar(**{'mass': 7.294196, 'state': 'H-rich_Core_H_burning', \
                                 'natal_kick_array': [57.65029, 5.359689, 1.86049, 6.200635]})
star_2 = SingleStar(**{'mass' : 7.028671, 'state' : 'H-rich_Core_H_burning'})
binary = BinaryStar(star_1, star_2,
                    **{'time': 0.0, 'state': 'detached', 'event': 'ZAMS', 'orbital_period': 77.926729, 'eccentricity': 0.0},
                    properties = sim_prop)

binary.evolve()

```

**Comment by celiotine** [comment]
Something that could be related, I noticed that the mass transfer case is not being set for these binaries at the necessary steps (it is `None`), even though they enter RLO. This could be why they are not correctly handled in `get_binary_state_and_event_and_mt_case()`. For example, in cases where they enter the "undefined" state at the CO_HMS_RLO step, the interpolation class for these binaries is being set to "unstable_MT", so not sure why the MT case would be `None`.

**Comment by astroJeff** [comment]
@celiotine hopefully PR#206 above will solve this issue.

**Comment by mkruckow** [comment]
@celiotine please check, that this issue is solved in the newest developement branch after we merged PR #206 . If the test is successful, please close the issue.
